qutebrowser/config/config.py

```diff
@@ -657,6 +672,50 @@ def set_command(self, win_id, section_=None, option=None, value=None,
             message.info(win_id, "{} {} = {}".format(
                 section_, option, val), immediately=True)
 
+    @cmdutils.register(instance='config')
+    def reload_config(self):
+        """Reload the config & keyconfig files."""
+        old_config = objreg.get('config')
+        objreg.delete('config')
+        try:
+            _init_main_config(QApplication.instance())
+        # Restore the old config if we failed
+        except:
```

Comment:
We never want to end up in a situation where we somehow end up with an invalid/incomplete config object in the objreg. I don't know which exceptions to expect, since there's a lot of code behind that one function, and any line may fail with any exception (due to a bug, unexpected scenario, etc.)